core/vm/evm.go

```diff
@@ -124,7 +138,17 @@ func NewEVM(ctx Context, statedb StateDB, chainConfig *params.ChainConfig, vmCon
 		chainRules:  chainConfig.Rules(ctx.BlockNumber),
 	}
 
-	evm.interpreter = NewInterpreter(evm, vmConfig)
+	if evm.interpreters == nil {
+		evm.interpreters = make([]Interpreter, 1, maxNumInterpreters)
+		evm.interpreters[0] = NewEVMInterpreter(evm, vmConfig)
+		evm.interpreter = evm.interpreters[0]
+	} else {
+		evm.interpreters = append(evm.interpreters, NewEVMInterpreter(evm, vmConfig))
+		if evm.interpreter == nil {
+			evm.interpreter = evm.interpreters[len(evm.interpreters)-1]
+		}
+	}
```

Comment:
This code is weird because `evm.interpreters` is always nil (the `EVM` is created just a few lines above).